pbj-core/pbj-runtime/src/main/java/com/hedera/pbj/runtime/io/buffer/Bytes.java

```diff
@@ -305,14 +305,52 @@ public void writeTo(@NonNull final MessageDigest digest, final int offset, final
         digest.update(buffer, Math.toIntExact(start + offset), length);
     }
 
+    /**
+     * Same as {@link #updateSignature(Signature, int, int)} with offset 0 and length equal to the length of this
+     * {@link Bytes} object.
+     */
+    public void updateSignature(@NonNull final Signature signature) throws SignatureException {
+        signature.update(buffer, start, length);
+    }
+
     /**
      * A helper method for efficient copy of our data into a Signature without creating a defensive copy of the data.
-     * The implementation relies on a well-behaved Signature that doesn't modify the buffer data.
+     * The implementation relies on a well-behaved Signature that doesn't modify the buffer data. Calls the
+     * {@link Signature#update(byte[], int, int)} method with all the data in this {@link Bytes} object. This method
+     * should be used when the data in the buffer should be validated or signed.
      *
-     * @param signature the Signature to copy into
+     * @param signature The Signature to update
+     * @param offset    The offset from the start of this {@link Bytes} object to get the bytes from
+     * @param length    The number of bytes to extract
+     * @throws SignatureException If the Signature instance throws this exception
      */
-    public void writeTo(@NonNull final Signature signature) throws SignatureException {
-        signature.update(buffer, start, length);
+    public void updateSignature(@NonNull final Signature signature, final int offset, final int length)
+            throws SignatureException {
+        signature.update(buffer, Math.toIntExact(start + offset), length);
```

Comment:
Minor: might want to validate the provided `offset` and `length` to ensure they are within this Bytes buffer's length and (start+length), and throw AIOOB if they aren't. The same comment for the `verifySignature` below, so perhaps a helper private `validateBounds` method is warranted here, which could be useful in other places in Bytes as well.

Reply:
I added such a method. It seems that it could be useful in other places as well